src/gov/nist/javax/sip/stack/SIPDialog.java

```diff
@@ -4409,8 +4409,17 @@ public void setEarlyDialogTimeoutSeconds(int seconds) {
     public void checkRetransmissionForForking(SIPResponse response) {
         final int statusCode = response.getStatusCode();
         final String responseMethod = response.getCSeqHeader().getMethod();
-        final long responseCSeqNumber = response.getCSeq().getSeqNumber();   
-        boolean isRetransmission = !responsesReceivedInForkingCase.add(statusCode + "/" + responseCSeqNumber + "/" + responseMethod);            
+        final long responseCSeqNumber = response.getCSeq().getSeqNumber();
+        /*From RFC3262, 1 Introduction: Each provisional response is given a sequence number, carried in the
+        RSeq header field in the response.  The PRACK messages contain an
+        RAck header field, which indicates the sequence number of the
+        provisional response that is being acknowledged.*/
+        RSeq rseq = (RSeq) response.getHeader(RSeqHeader.NAME);
```

Comment:
this looks much cleaner, and with proper comment. Could to rework the PR so there is just one commit with latest changes?